File: Lab5/mpj_hello_world/src/main/java/org/example/MatrixFunctions.java
package org.example;

public class MatrixFunctions {
    public static int[][] Multiply(int[][] matrixA, int[][] matrixB) {
        var result = new int[matrixA.length][matrixA[0].length];

        for (int i = 0; i < matrixA.length; i++) {
            for (int j = 0; j < matrixA[0].length; j++) {
                for (int k = 0; k < matrixA[0].length; k++) {
                    result[i][j] += matrixA[i][k] * matrixB[k][j];
                }
            }
        }

        return result;
    }

    public static int[][] GetSubMatrix(int offset, int rows, int[][] matrix) {
        int columns = matrix[0].length;
        var subMatrix = new int[rows][columns];

        for (int i = 0; i < rows; i++) {
            subMatrix[i] = matrix[offset + i];
        }

        return subMatrix;
    }
}

File: Lab8/MatrixMultiplicationWebApp/MatrixMultiplicationWebApp.Core/MatrixFunctions.cs
namespace MatrixMultiplicationWebApp.Core;

public static class MatrixFunctions
{
    public static int[][] GetTransposed(int [][] matrix)
    {
        var rows = matrix.Length;
        var columns = matrix[0].Length;

        var transposedMatrix = new int[columns][];

        for(int i = 0; i < rows; i++)
        {
            transposedMatrix[i] = new int[rows];
            for (int j = 0; j < columns; j++)
            {
                transposedMatrix[j][i] = matrix[i][j];
            }
        }

        return transposedMatrix;
    }
}